packit_config_checker.py

```diff
@@ -255,6 +262,61 @@ def commit_file_in_new_branch_github(self):
 
         return True
 
+    def commit_file_in_new_branch_pagure(self):
+        commit_sha = self.project.get_sha_from_branch(self.project.default_branch)
+        fork_commit_sha = self.fork.get_sha_from_branch(self.fork.default_branch)
+
+        directory = tempfile.mkdtemp()
+        url = self.fork.get_git_urls()["ssh"]
+        git_repo = git.repo.Repo.clone_from(url=url, to_path=directory, tags=True)
+        origin = git_repo.remote("origin")
+
+        if commit_sha != fork_commit_sha:
+            click.echo(
+                "Fork is not synced with the parent repo"
+            )
+            upstream = git_repo.create_remote(
+                name="upstream",
+                url=self.project.get_git_urls()["ssh"],
+            )
+            upstream.fetch()
+            branch_name = git_repo.active_branch.name
+            head = git_repo.heads[branch_name]
+            upstream_head = upstream.refs[branch_name]
+            head.set_commit(upstream_head)
+            origin.push(refspec=head)
+
+
+        parent_default_branch = [branch for branch in git_repo.branches 
+                                 if branch.name == self.project.default_branch][0]
+        parent_commit = parent_default_branch.commit
+
+        refs = [
+            branch
+            for branch in git_repo.branches
+            if branch.name == SOURCE_BRANCH
+        ]
+        if refs:
+            ref = refs[0]
+        else:
+            ref = git_repo.create_head(SOURCE_BRANCH, parent_commit)
+
+        git_repo.head.reference = ref
+        git_repo.head.reset(index=True, working_tree=True)
+
+        file = os.path.join(git_repo.working_tree_dir, self.config_file_name)
+        with open(file, "w") as f:
+            f.write(self.updated_packit_config)
+        git_repo.index.add([file])
+        git_repo.index.commit(self.commit_msg)
+        try:
+            git_repo.git.execute(["git", "push", "-u", "origin", SOURCE_BRANCH])
```

Comment:
Why calling external git binary?

```suggestion
            git_repo.remote(name="origin").push(SOURCE_BRANCH)
```

Reply:
Because I was not able to find a way to track a *not yet created origin repo* using the library... There is an example but it is using an already existing origin branch for tracking. I decided I had already spent to much time on it 😅

Reply:
That's probably impossible, but `origin` is autocreated after cloning, right? And you are already ccalling `origin.push()` a few lines above.

Reply:
Yes, but few lines above I was pushing to the default origin branch.
Here I have created a new branch named `SOURCE_BRANCH` and I need to track it on the origin where it does not yet exist... I tried in a couple of ways and I failed so I gave up using the API.

Reply:
Oh, you meant not-yet-created branch in the origin remote. Well, I tried that with the proposed code and it just works.

Reply:
As for the tracking, I haven't tried that, but let me have a look.

Reply:
This should do it:

```suggestion
            git_repo.git.execute(["git", "push", "-u", "origin", SOURCE_BRANCH])
            origin.push(refspec=SOURCE_BRANCH)
            git_repo.head.reference.set_tracking_branch(getattr(origin.refs, SOURCE_BRANCH))
```

Probably there is a more elegant way than using `getattr()`, but this works :slightly_smiling_face: